README.md

```diff
@@ -1,133 +1,616 @@
+  ```java
+  // src/main/java/typescript/example/client/gradle/ExampleClientEndpointResolutionIntegration.java
+  package typescript.example.client.gradle;
+  
+  // ...
+  
+  public class ExampleClientEndpointResolutionIntegration implements TypeScriptIntegration {
+      @Override
+      public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
+          TypeScriptSettings settings,
+          Model model,
+          SymbolProvider symbolProvider,
+          LanguageTarget target
+      ) {
+          // Runtime config value also be specified per platform by using the `target` argument, e.g.
+          // if (target.equals(LanguageTarget.NODE)) { ... }
+  
+          // This example provides an arbitrary endpoint provider based on random numbers
+          return Map.of("endpoint", w -> w.write("""
+              async () => {
+                // Generate number between 1-10
+                const n = Math.floor(Math.random() * 11);
+                // Return domain based on random number
+                return n < 5
+                  ? "https://www.example.com"
+                  : "https://www.amazon.com";
+              }"""));
+      }
+  }
+  ```
+- Registering the `TypeScriptIntegration`:
+  ```java
+  // src/main/resources/META-INF/services/software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration
+  typescript.example.client.gradle.ExampleClientEndpointResolutionIntegration
+  ```
+
+See [the section on customizations via `TypeScriptIntegration` for more details](#customizations-via-typescriptintegration).
+
+#### Publishing a Client SDK package
+
+> Note: There is no prescribed way to publish NPM packages since there are many ways to maintain SDKs. Some publishing tools include using [`npm publish`](https://docs.npmjs.com/cli/v8/commands/npm-publish) or [`yarn publish`](https://classic.yarnpkg.com/lang/en/docs/cli/publish/) directly, or managing a monorepo with tools like [`turbo`](https://turbo.build/repo/docs/handbook/publishing-packages). This section provides tips for how a general publishing workflow could work.
+
+A generated client is a package that is ready to be published. After running `smithy build`, the generated client artifacts will be in the build directory under the projection and plugin name. For example, generated client artifacts for the source projection using the `typescript-client-codegen` plugin in a Smithy CLI project would be in the `build/smithy/source/typescript-client-codegen/` directory. A common practice is to copy the generated client artifacts into a source control repository. After the artifacts are staged, any modifications that are needed prior to publishing the generated client artifacts specific to the SDK should be run, e.g. adding a `README.md`, editing changelog entries. Finally, with a chosen publishing tool for the SDK, publish the artifacts after running the `prepack` script per client package.
+
+#### `typescript-client-codegen` plugin configuration
```

Comment:
I think it might be clearer to move these sections up to right after we introduce each plugin.

Reply:
Moved the client configuration section up first under the Client SDK plugin section.